scylla-cql/src/frame/response/result.rs

```diff
@@ -730,21 +746,30 @@ fn mk_col_spec_parse_error(
 
 fn deser_col_specs(
     buf: &mut &[u8],
-    global_table_spec: &Option<TableSpec<'static>>,
+    global_table_spec: Option<TableSpec<'static>>,
     col_count: usize,
 ) -> StdResult<Vec<ColumnSpec<'static>>, ColumnSpecParseError> {
+    let global_table_spec_provided = global_table_spec.is_some();
+    let mut table_spec = global_table_spec;
+
     let mut col_specs = Vec::with_capacity(col_count);
     for col_idx in 0..col_count {
-        let table_spec = if let Some(spec) = global_table_spec {
-            spec.clone()
-        } else {
-            deser_table_spec(buf).map_err(|err| mk_col_spec_parse_error(col_idx, err))?
-        };
+        if !global_table_spec_provided {
+            table_spec = Some(
+                deser_table_spec(buf, table_spec)
+                    .map_err(|err| mk_col_spec_parse_error(col_idx, err))?
+                    .to_owned(),
+            );
+        }
         let name = types::read_string(buf)
             .map_err(|err| mk_col_spec_parse_error(col_idx, err))?
             .to_owned();
         let typ = deser_type_owned(buf).map_err(|err| mk_col_spec_parse_error(col_idx, err))?;
-        col_specs.push(ColumnSpec::owned(name, typ, table_spec));
+        col_specs.push(ColumnSpec::owned(
+            name,
+            typ,
+            table_spec.as_ref().unwrap().clone(),
```

Comment:
nit: justify unwrap

Reply:
+1, it's not obvious to me that `col_count` must be greater than 0.

Reply:
Good that you noticed that bad practice. I refactored the code much.